dev-games/ogre/ogre-1.11.1.ebuild

```diff
@@ -0,0 +1,138 @@
+	+freeimage gles2 json ois openexr +opengl pch profile +resman-legacy
+	resman-pedantic resman-strict tools"
+
+REQUIRED_USE="
+	|| ( gles2 opengl )
+	^^ ( resman-legacy resman-pedantic resman-strict )
+	examples? ( ois )"
+
+RESTRICT="test" #139905
+
+RDEPEND="
+	dev-libs/zziplib
+	media-libs/freetype:2
+	x11-libs/libX11
+	x11-libs/libXaw
+	x11-libs/libXrandr
+	x11-libs/libXt
+	cg? ( media-gfx/nvidia-cg-toolkit )
+	egl? ( media-libs/mesa[egl] )
+	freeimage? ( media-libs/freeimage )
+	gles2? ( media-libs/mesa[gles2] )
+	json? ( dev-libs/rapidjson )
+	ois? ( dev-games/ois )
+	openexr? ( media-libs/openexr )
+	opengl? (
+		virtual/glu
+		virtual/opengl
+	)"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	x11-base/xorg-proto
+	doc? ( app-doc/doxygen )"
+
+PATCHES=(
+	"${FILESDIR}/${P}-media_path.patch"
+	"${FILESDIR}/${P}-resource_path.patch"
+	"${FILESDIR}/${P}-samples.patch"
+	"${FILESDIR}/${P}-fix_sample_source_install.patch"
+)
+
+src_prepare() {
+	sed -i \
+		-e "s:share/OGRE/docs:share/doc/${PF}:" \
+		Docs/CMakeLists.txt || die
+	# Stupid build system hardcodes release names
+	sed -i \
+		-e '/CONFIGURATIONS/s:CONFIGURATIONS Release.*::' \
+		CMake/Utils/OgreConfigTargets.cmake || die
+
+	# Fix some path issues
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DOGRE_BUILD_COMPONENT_BITES=yes
+		-DOGRE_BUILD_COMPONENT_HLMS=$(usex beta-components)
+		-DOGRE_BUILD_COMPONENT_JAVA=no
+		-DOGRE_BUILD_COMPONENT_PAGING=yes
+		-DOGRE_BUILD_COMPONENT_PROPERTY=yes
+		-DOGRE_BUILD_COMPONENT_PYTHON=no
+		-DOGRE_BUILD_COMPONENT_RTSHADERSYSTEM=yes
+		-DOGRE_BUILD_COMPONENT_TERRAIN=yes
+		-DOGRE_BUILD_COMPONENT_VOLUME=yes
+		-DOGRE_BUILD_DEPENDENCIES=no
+		-DOGRE_BUILD_PLUGIN_CG=$(usex cg)
+		-DOGRE_BUILD_PLUGIN_FREEIMAGE=$(usex freeimage)
+		-DOGRE_BUILD_PLUGIN_EXRCODEC=$(usex openexr)
+		-DOGRE_BUILD_RENDERSYSTEM_GL=$(usex opengl)
+		-DOGRE_BUILD_RENDERSYSTEM_GL3PLUS=$(usex opengl)
+		-DOGRE_BUILD_RENDERSYSTEM_GLES2=$(usex gles2)
+		-DOGRE_BUILD_SAMPLES=$(usex examples)
+		-DOGRE_BUILD_TESTS=no
+		-DOGRE_BUILD_TOOLS=$(usex tools)
+		-DOGRE_CONFIG_DOUBLE=$(usex double-precision)
+		-DOGRE_CONFIG_ENABLE_GL_STATE_CACHE_SUPPORT=$(usex cache)
+		-DOGRE_CONFIG_ENABLE_GLES2_CG_SUPPORT=$(usex gles2 $(usex cg) no)
+		-DOGRE_CONFIG_ENABLE_GLES3_SUPPORT=$(usex gles2)
+		-DOGRE_CONFIG_THREADS=3
+		-DOGRE_CONFIG_THREAD_PROVIDER=std
+		-DOGRE_ENABLE_PRECOMPILED_HEADERS=$(usex pch)
+		-DOGRE_FULL_RPATH=no
+		-DOGRE_GLSUPPORT_USE_EGL=$(usex egl)
+		-DOGRE_INSTALL_DOCS=$(usex doc)
+		-DOGRE_INSTALL_SAMPLES=$(usex examples)
+		-DOGRE_INSTALL_SAMPLES_SOURCE=$(usex examples)
+		-DOGRE_PROFILING=$(usex profile)
+		-DOGRE_RESOURCEMANAGER_STRICT=$(\
+			usex resman-pedantic 1 $(\
+			usex resman-strict 2 0))
+	)
+
+	# Ogre-1.11+ is making use of "CMAKE_INSTALL_CONFIG_NAME MATCHES ..."
+	# and sets it to BUILD_TYPE. Only RelWithDebInfo, MinSizeRel and Debug
+	# are supported.
+	CMAKE_BUILD_TYPE="$(usex debug Debug RelWithDebInfo)"
```

Comment:
From `cmake-utils.eclass`:
```
# @ECLASS-VARIABLE: CMAKE_BUILD_TYPE
# @DESCRIPTION:
# Set to override default CMAKE_BUILD_TYPE. Only useful for packages
# known to make use of "if (CMAKE_BUILD_TYPE MATCHES xxx)".
# If about to be set - needs to be set before invoking cmake-utils_src_configure.
# You usualy do *NOT* want nor need to set it as it pulls CMake default build-type
# specific compiler flags overriding make.conf.
: ${CMAKE_BUILD_TYPE:=Gentoo}
```

Is that something you can fix with a build system patch instead? What is it used for?

Reply:
...I seem to be unable to find out where it comes from, but while searching I found out that ogre-1.10.12 does it like that, too.

Reply:
Okay, ogre-1.10.x does not need this. But ogre-1.11.x does not install any binaries if this is removed.

However, instead of setting CMAKE_BUILD_TYPE, I am currently testing to simply define CMAKE_INSTALL_CONFIG_NAME to an appropriate value. I _think_ that variable is only set to BUILD_TYPE, if it has not been defined before.

Reply:
I am sorry, but this does not work. I have tried all different sorts of things, but the build system insists on "`Install Type : Gentoo`" whenever CMAKE_BUILD_TYPE is not manipulated in src_configure.
And the result is, that neither binaries nor libraries get installed.

I have no idea how this gets added to the cmake_install.cmake and thus can not patch it, sorry. I am absolutely no cmake pro, and google failed at this one.

Reply:
ogre-2.1 doesn't need this either. It seems to be particular for the 1.11 series.